### PR TITLE
Add transient program

### PR DESCRIPTION
**Pull request by andywingo**
This load tester is configured as a Snabb application, so it can be run as "snabb transient ...".  Like the lwaftr loadgen tester, it writes rate-limited data to a NIC and measures what it gets back in return.  It prints resulting statistics in CSV format.  Unlike the lwaftr loadgen tester it uses timers and the link statistics instead of a statistics app, and produces a load transient instead of just one rate.

This program is unlike the packetblaster in that it limits its rate, produces a load transient, and measures return traffic, but also because it uses the low-latency Intel82599 app instead of the intel.LoadGen app.

### CODE REVIEW

File: src/program/transient/transient.lua [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+local config    = require("core.config")
+local timer     = require("core.timer")
+local pci       = require("lib.hardware.pci")
+local Intel82599= require("apps.intel.intel_app").Intel82599
+local basic_apps = require("apps.basic.basic_apps")
+local main      = require("core.main")
+local PcapReader= require("apps.pcap.pcap").PcapReader
+local lib = require("core.lib")
+local ffi = require("ffi")
+local C = ffi.C
```
Comment (dpino) [inline src/program/transient/transient.lua]: `C` is not used. The headers seem to be column-aligned but some are not. It's not a requirement of SnS to column-align headers but it's strange to see it a new file.

File: src/program/transient/transient.lua [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+   local devices = find_devices(pattern)
+   if #devices == 0 then
+      error('no devices matched pattern "'..pattern..'"')
+   elseif #devices == 1 then
+      return devices[1]
+   else
+      local devices_str = ''
+      for _,device_str in ipairs(devices) do
+         devices_str = devices_str .. ' ' .. device_str
+      end
```
Comment (dpino) [inline src/program/transient/transient.lua]: Same as 
```lua
local devices_str = table.concat(devices, " ")
```

File: src/program/transient/transient.lua [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+   function handlers.s(arg)
+      opts.step = assert(tonumber(arg), 'step must be a number')
+   end
+   function handlers.D(arg)
+      opts.duration = assert(tonumber(arg), 'duration must be a number')
+   end
+   function handlers.p(arg)
+      opts.period = assert(tonumber(arg), 'period must be a number')
+   end
+   function handlers.h(arg) show_usage(0) end
```
Comment (dpino) [inline src/program/transient/transient.lua]: It's not necessary to define `arg` unless the function expects a parameter.

File: src/program/transient/transient.lua [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+      end
+      table.insert(statistics, sample)
+      print_sample(sample)
+   end
+end
+
+function run(args)
+   local opts, streams = parse_args(args)
+   local c = config.new()
+   for i, stream in ipairs(streams) do
```
Comment (dpino) [inline src/program/transient/transient.lua]: `i` is not used.

**Comment by dpino** [comment]
LGTM.

**Comment by andywingo** [comment]
Fixed nits, thanks for the review!

**Pull request merged**